File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcStructuralActivity/IfcStructuralReaction/IfcStructuralSurfaceReaction/IfcStructuralSurfaceReaction.py
class IfcStructuralSurfaceReaction(IfcStructuralReaction):
    def __init__(self):
        self.PredefinedType: IfcStructuralSurfaceActivityTypeEnum = None

    # === WHERE CLAUSES ===
    # HasPredefinedType : (PredefinedType <> IfcStructuralSurfaceActivityTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcStructuralActivity/IfcStructuralReaction/IfcStructuralSurfaceReaction/IfcStructuralSurfaceReaction.js
class IfcStructuralSurfaceReaction extends IfcStructuralReaction {
    constructor() {
        /** @type {IFCSTRUCTURALSURFACEACTIVITYTYPEENUM} */
        this.PredefinedType = null;
    }

    // === WHERE CLAUSES ===
    // HasPredefinedType : (PredefinedType <> IfcStructuralSurfaceActivityTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
}
